lib/sales_analyst.rb

```diff
@@ -1,71 +1,112 @@
-    @engine = engine
-    @sales_analyst = sales_analyst
+  def initialize(sales_engine)
+    @sales_engine = sales_engine
+    # @sales_analyst = sales_analyst
   end
 
   def average_items_per_merchant
-    sales_engine.items.items.each do |item|
-      require 'pry'; binding.pry
-      items.count
-      require 'pry'; binding.pry
-    end
-    require 'pry'; binding.pry
+   (items_count / merchants_count.to_f).round(2)
     # total number of items per merchant
     # divided by total number of merchants
-    # Float
   end
 
-  def average_items_per_merchant
-    # total number of items per merchant
-      # add up the sums 
-    # divided by total number of merchants
-    # Float
+  def items_count
+    sales_engine.items.all.count
+  end
+
+  def merchants_count
+    sales_engine.merchants.all.count
   end
 
   def average_items_per_merchant_standard_deviation
+    sum = 0
+    sales_engine.merchants.all.each do |merchant|
+      sum += (sales_engine.items.find_all_by_merchant_id(merchant.id).count - average_items_per_merchant)**2
+    end
+    Math.sqrt(sum / (merchants_count - 1)).round(2)
     # take average_items_per_merchant 
     # find the standard deviation
         # (item_count_ForEachMerchant - average_items_per_merchant) ^ 2
         # add all of these up for every single merchant = sum
             #  sum / (by total number of merchants - 1) = answer
         # sqrt(answer) = sd
     # Float
-
   end
 
   def merchants_with_high_item_count
-    # use average_items_per_merchant_standard_deviation 
-    # find_all merchants with more than ONE sd ABOVE the average number of items for sale
+    high_item_count = sales_engine.merchants.all.find_all do |merchant|
+      (sales_engine.items.find_all_by_merchant_id(merchant.id).count) > (average_items_per_merchant + average_items_per_merchant_standard_deviation)
+    end
+    # find_all merchants with more than ONE sd ABOVE the average number of items offered
     # return array of merchants
   end
 
-  def average_item_price_for_specific_merchant(merchant_id)
-    # total of all item unit_price per merchant(use merchant ID)
-    # divide by total number of items for THAT merchant
-    # BigDecimal
-  end
+  def average_item_price_for_merchant(merchant_id)
+    item_prices_per_merchant = []
+      items_per_merchant = sales_engine.items.find_all_by_merchant_id(merchant_id)
+      items_per_merchant.each do |item|
+        item_prices_per_merchant << (item.unit_price / 100)
+      end
+      avg = ((item_prices_per_merchant.sum) / items_per_merchant.count)
+      avg = BigDecimal(avg, 4)
+      # price of each item for that merchant
+      # add it all together
+      # divide by number of items for THAT merchant
+        # divide ^ by 100 because items are in cents and we want dollars
+      #BigDecimal
+    end
 
-  def average_item_price_for_all_merchants
-    # use average_item_price_for_specific_merchant(merchant_id)
+  def average_average_price_per_merchant
```

Comment:
We'll want to double check the method names are the right method names the spec harness is looking for